### PR TITLE
Remove deprecated created and found from index, delete and bulk

### LINKED ISSUE #19630: Remove found and created from index, delete, update, and bulk
People can just use `operation` instead.

### PR DESCRIPTION
**Pull request by olcbean**
The `created` and `found` fields in index and delete responses became obsolete after the introduction of the `result` field in index, update and delete responses (#19566).

After deprecating the `created` and `found` fields in 5.x (#19633), now they are removed.

Fixes #19630

### CODE REVIEW

File: core/src/main/java/org/elasticsearch/action/delete/DeleteResponse.java [hunk 1]
```diff
@@ -104,15 +85,10 @@ public static void parseXContentFields(XContentParser parser, Builder context) t
...
-        private boolean found = false;
-
-        public void setFound(boolean found) {
-            this.found = found;
-        }
-
         @Override
         public DeleteResponse build() {
-            DeleteResponse deleteResponse = new DeleteResponse(shardId, type, id, seqNo, primaryTerm, version, found);
+            DeleteResponse deleteResponse = new DeleteResponse(shardId, type, id, seqNo, primaryTerm, version,
```
Comment (nik9000) [inline core/src/main/java/org/elasticsearch/action/delete/DeleteResponse.java]: I think you should pass the Result here. Maybe also add an assertion in the ctor that the result is either `Result.DELETED` or `Result.NOT_FOUND`.

File: core/src/main/java/org/elasticsearch/action/index/IndexResponse.java [hunk 2]
```diff
@@ -107,15 +88,10 @@ public static void parseXContentFields(XContentParser parser, Builder context) t
...
-
-        public void setCreated(boolean created) {
-            this.created = created;
-        }
-
         @Override
         public IndexResponse build() {
-            IndexResponse indexResponse = new IndexResponse(shardId, type, id, seqNo, primaryTerm, version, created);
+            IndexResponse indexResponse = new IndexResponse(shardId, type, id, seqNo, primaryTerm, version,
+                    result == Result.CREATED ? true : false);
```
Comment (nik9000) [inline core/src/main/java/org/elasticsearch/action/index/IndexResponse.java]: Same deal here as on Delete. I think you should pass the `result` into the ctor and `assert` that it is sane.

File: docs/reference/docs/bulk.asciidoc [hunk 3]
```diff
@@ -138,7 +136,6 @@ The result of this bulk operation is:
                "successful": 1,
                "failed": 0
             },
-            "created": true,
```
Comment (nik9000) [inline docs/reference/docs/bulk.asciidoc]: I'm so glad we assert that these are right now!

Comment (nik9000) [inline docs/reference/docs/bulk.asciidoc]: This is exactly the kind of thing I'd miss when removing these without the assertions.

**Review by nik9000 — CHANGES_REQUESTED** [review]
I left two minor things. Otherwise I think this looks good!

Thanks for working on it @olcbean!

**Comment by nik9000** [comment]
Marked as `breaking` because it is but that is ok because it removes deprecated response parameters that have had an alternate way to read them for most of the 5.x series.

**Comment by olcbean** [comment]
@nik9000 thanks for the feedback! Just pushed a new commit

**Comment by nik9000** [comment]
Ah! I see now. The way you had it first was better. I'll revert the second commit and merge.

**Comment by nik9000** [comment]
Thanks @olcbean! I've merged to master.

**Comment by olcbean** [comment]
@nik9000 Excellent, this is what I was hoping the final change to look like. Sorry about the confusion with the belated push earlier :)

**Pull request merged**